source/server/debug_session_properties_restricted_service.cpp

```diff
@@ -0,0 +1,176 @@
+{
+  return set_bool_property(context, (NISysCfgResourceProperty)0x10003000, request->device_id(), request->out_of_process());
+}
+
+::grpc::Status DebugSessionPropertiesRestrictedService::access_syscfg_resource_by_device_id_filter(
+  ::grpc::ServerContext* context,
+  const nidevice_restricted_grpc::DeviceId& device_id,
+  std::function<NISysCfgStatus(nidevice_grpc::SysCfgLibraryInterface*, NISysCfgResourceHandle, bool*)> syscfg_resource_action_func)
+{
+  NISysCfgStatus status = NISysCfg_OK;
+  NISysCfgSessionHandle session = NULL;
+  NISysCfgFilterHandle filter = NULL;
+  NISysCfgEnumResourceHandle resources_handle = NULL;
+  NISysCfgResourceHandle resource = NULL;
+
+  try {
+    auto library = get_syscfg_library_interface();
+    if (NISysCfg_Succeeded(status = open_or_get_localhost_syscfg_session(&session))) {
+      if (NISysCfg_Succeeded(status = library->CreateFilter(session, &filter))) {
+        library->SetFilterProperty(filter, NISysCfgFilterPropertyIsDevice, NISysCfgBoolTrue);
+        library->SetFilterProperty(filter, NISysCfgFilterPropertyUserAlias, device_id.name().c_str());
+        library->SetFilterProperty(filter, NISysCfgFilterPropertySerialNumber, device_id.serial_number().c_str());
+        library->SetFilterProperty(filter, NISysCfgFilterPropertyProductId, device_id.product_id());
+        if (NISysCfg_Succeeded(status = library->FindHardware(session, NISysCfgFilterModeMatchValuesAll, filter, NULL, &resources_handle))) {
+          if (NISysCfg_Succeeded(status) && (status = library->NextResource(session, resources_handle, &resource)) == NISysCfg_OK) {
+            bool save_changes = false;
+            if (NISysCfg_Succeeded(status = syscfg_resource_action_func(library, resource, &save_changes))) {
+              if (save_changes) {
+                NISysCfgBool changes_require_restart = NISysCfgBoolFalse;
+                char* detailed_changes = nullptr;
+                status = library->SaveResourceChanges(resource, &changes_require_restart, &detailed_changes);
+                library->FreeDetailedString(detailed_changes);
+              }
+            }
+            library->CloseHandle(resource);
+          }
+          library->CloseHandle(resources_handle);
+        }
+        library->CloseHandle(filter);
+      }
+    }
+  }
+  catch (nidevice_grpc::NonDriverException& ex) {
+    return ex.GetStatus();
+  }
+
+  if (NISysCfg_Failed(status)) {
+    return ::grpc::Status(::grpc::StatusCode::INTERNAL, kDebugSessionPropertyAccessFailedMessage);
+  }
+
+  return ::grpc::Status::OK;
+}
+
+::grpc::Status DebugSessionPropertiesRestrictedService::get_bool_property(
+  ::grpc::ServerContext* context,
+  NISysCfgResourceProperty property_id,
+  const nidevice_restricted_grpc::DeviceId& device_id,
+  bool* value)
+{
+  auto get_bool_lambda = [&] (nidevice_grpc::SysCfgLibraryInterface* library, NISysCfgResourceHandle resource, bool* save_changes) {
+    *save_changes = false;
```

Comment:
Since it defaults to false, resetting it here seems superfluous.

Reply:
Removed unnecessary assignment.